src/pip_requirements_parser.py

```diff
@@ -264,16 +265,10 @@ def from_string(cls, text: str) -> "RequirementsFile":
         Since pip requirements are deeply based on files, we create a temp file
         to feed to pip even if this feels a bit hackish.
         """
-        tmpdir = None
-        try:
-            tmpdir = Path(str(tempfile.mkdtemp()))
-            req_file = tmpdir / "requirements.txt"
-            with open(req_file, "w") as rf:
-                rf.write(text)
-            return cls.from_file(filename=str(req_file), include_nested=False)
-        finally:
-            if tmpdir and tmpdir.exists():
-                shutil.rmtree(path=str(tmpdir), ignore_errors=True)
+        with NamedTemporaryFile("w") as tf:
```

Comment:
:x: this must be "w+"

otherwise, this will crash on windows systems. ... 
read https://stackoverflow.com/questions/23212435/permission-denied-to-write-to-my-temporary-file

Reply:
please add a proper unit test for this method.
need to check especially how python behaves on windws envs.

Reply:
found this one: https://github.com/python/cpython/issues/58451

Reply:
so i would prefer the fowllong: 
```py
from os import unlink
tf = NamedTemporaryFile('wt', delete=False,
                        # we prefer utf8 encoded strings, but ...
                        # - must not change newlines
                        # - must not  change encoding, fallback to system encoding for compatibility
                        newline='', encoding=None)
tf.write(text)
tf.close()
try:
  return cls.from_file(filename=tf, include_nested=False)
finally:
  unlink(tf)
```


this worked on windows and linux in my tests

Reply:
Force pushed that change. 5 tests still failing, but they are also failing in the same way on `main`